server/neptune/gateway/event/deployer.go

```diff
@@ -51,10 +54,12 @@ type RootDeployOptions struct {
 	RepoFetcherOptions *github.RepoFetcherOptions
 	Trigger            workflows.Trigger
 	Rerun              bool
+
+	// only used in force apply comments (if a force apply comment contains legacy mode roots)
+	InitialRequest *http.BufferedRequest
```

Comment:
Why did you modify something that's specific to neptune? This is deployer is supposed to have no knowledge of legacy things and by doing this, we're making it harder to remove legacy code since there's tests that are impacted etc.

Can't you just copy what we're doing in the autoplanner in the comment handler to generate root config and forward to sns?

We should keep this code separate to minimize any risk to neptune as it stands this seems like push events are going to trigger applies now?